fractal_server/app/api/v1/workflow.py

```diff
@@ -83,3 +86,55 @@ async def delete_workflow(
     await db.commit()
 
     return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@router.get("/{_id}", response_model=WorkflowRead)
+async def get_workflow(
+    _id: int,
+    user: User = Depends(current_active_user),
+    db: AsyncSession = Depends(get_db),
+):
+    workflow = db.get(Workflow, _id)
+    if not workflow:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Workflow not found"
+        )
+    # check autorization
+    await get_project_check_owner(
+        project_id=workflow.project_id,
+        user_id=user.id,
+        db=db,
+    )
+    return workflow
+
+
+@router.patch("{_id}/add-task/", response_model=WorkflowTaskRead)
```

Comment:
sorry, this is really my bad, let's change this to POST. It is true that we are modifying the workflow, but we are most of all creating a WorkflowTask (return 201).

Also, in this case it is better to return the whole Workflow, as if we insert in the middle (`order != len(Workflow.task_list)`) it would be difficult for the clients to use the information of the task alone, as all the following tasks would change index.